designs/gpio-plus/README.md

```diff
@@ -0,0 +1,231 @@
+
+1. Add advanced mode checkbox. 
+2. In advanced mode, the following values can be received as a payload value instead of simple integer value:
+
+  a. JavaScript object containing following properties:
+  
+  | Name    | Type  | required  | Desc.                    |
+  |:--------|:------|:----------|:-------------------------|
+  | `pin`   | int   | +         | GPIO pin number          |
+  | `value` | number| -         | value to output          |
+  | `delay` | number| -         | delay after output in ms |
+  | `wait`  | number| -         | wait befor output in ms  |
+
+   These property values are used to output values to GPIO pins.
+  
+  - Example: output `1` to pin#11 then wait for 100ms
+
+  ```[javascript]
+  {
+      pin: 11,
+      value: 1
+      delay: 100
+  }
+  ```
+
+  b. `pin` of (a) can be array of GPIO pin numbers.  In this case, value specified by `value` is outputted to specified GPIO pins.
+  
+  - Example: output `1` to pin#11 and pin#12 then wait for 100ms
+
+  ```[javascript]
+  {
+      pin: [11, 12]
+      value: 1
+      delay: 100
+  }
+  ```
+
+  c. `pin` and  `value` of (a) can be arrays of same length.  In this case,  a value in `value` array is outputted to corresponding pin in `pin` array.
+  
+  - Example: output `1` to pin#11 and `0` to pin#12 then wait for 100ms
+
+  ```[javascript]
+  {
+      pin: [11, 12]
+      value: [1, 0]
+      delay: 100
+  }
+  ```
+
+  d. `pin` of (a) can point to a object that contains pin number as a key and corresponding GPIO input value as its value is sent as payload value.
+  
+  - Example: output `1` to pin#11 and `0` to pin#12 then wait for 100ms
+
+  ```[javascript]
+  {
+      pin: {
+        '11': 1,
+        '12': 0
+      },
+      delay: 100
+  }
+  ```
+  
+  d. array of object of (a) ~ (c).  In this case, the objects in the array are applied sequentially.
+
+  - Example: output `1` to pin#11, wait for 100ms, then output `0` to pin#12
+
+  ```[javascript]
+  [
+      {
+          pin: 11,
+          value: 1,
+          delay: 100
+      },
+      {
+          pin: 12,
+          value: 0
+      }
+  ]
+  ```
+  
+  e. JavaScript object containing following properties:
+
+  | Name       | Type  | required  | Desc.           |
+  |:-----------|:------|:----------|:----------------|
+  | `pin`      | int   | +         | GPIO pin number |
+  | `mode`     | string| +         | pmw/digital     |
+  | `frequency`| number| -         | for PWM mode    |
+  
+  These properties are used to setup mode of a GPIO output pin.
+  In PWM mode, output value of GPIO pin can be floating point or `null`/`undefined`.  If the values is `null`/`undefined`, PWM for specified GPIO pin is stopped.
+
+  - Example: setup pin#11 for PWM mode for 5hz
+  
+  ```[javascript]
+  {
+      pin: 11,
+      mode: "pwm",
+      frequency: 5
+  }
+  ```  
+  
+3. In advanced mode, **GPIO out** node has an output port.  The **GPIO out** node outputs a message containing payload with `true` value.
```

Comment:
Is this the same as done ? ie triggered at end of a sequence... ?

Reply:
Yes. Since the flow using `done` is not easy to understand the execution ordering of nodes, I want to make the connection relation of the nodes explicit.